process/block/preprocess/transactions.go

```diff
@@ -746,6 +736,19 @@ func (txs *transactions) ProcessMiniBlock(
 		if err != nil {
 			return err
 		}
+
+		processedTxHashes = append(processedTxHashes, miniBlockTxHashes[index])
+	}
+
+	for index := range miniBlockTxs {
```

Comment:
i don't think we need a new for. It could stay in the first for cycle. the code would be easier as well.

Reply:
If we move this inside the next for there could be an edge case. Actually this method is called only for crossshard miniblocks from processCompleteMiniBlock method, when a node is proposer. If let say that this miniblock is send by a malicious consensus group from another shard and it would have gas limit consumption over the max limit, but this limit would be detected only after the node would processed some txs from this miniblock, the return from this method it would let the accumulatedFees dirty. From this point even if this proposer would skip this bad miniblock and it would take another one the accumulatedFees would be wrongly calculated on its side compared with what would be calculated on its consensus group side.

Reply:
ok. can remain as this.